bsp4s/src/main/scala/ch/epfl/scala/bsp/Bsp.scala

```diff
@@ -75,54 +84,131 @@ object BuildTargetTag {
     languageIds: List[String],
     dependencies: List[BuildTargetIdentifier],
     dataKind: Option[String],
-    data: Option[Json]
+    data: Option[RawJson]
 )
 
+object BuildTarget {
+  implicit val codec: JsonValueCodec[BuildTarget] =
+    JsonCodecMaker.makeWithRequiredCollectionFields
+}
+
 object BuildTargetDataKind {
   val Scala = "scala"
   val Sbt = "sbt"
 }
 
-@JsonCodec final case class BuildClientCapabilities(
+final case class BuildClientCapabilities(
     languageIds: List[String]
 )
 
+object BuildClientCapabilities {
+  implicit val codec: JsonValueCodec[BuildClientCapabilities] =
+    JsonCodecMaker.makeWithRequiredCollectionFields
+}
+
 // Notification: 'build/initialized', C -> S
-@JsonCodec final case class InitializedBuildParams()
+final case class InitializedBuildParams()
+
+object InitializedBuildParams {
+  implicit val codec: JsonValueCodec[InitializedBuildParams] =
+    JsonCodecMaker.makeWithRequiredCollectionFields
+}
 
 // Request: 'build/initialize', C -> S
-@JsonCodec final case class InitializeBuildParams(
+final case class InitializeBuildParams(
     displayName: String,
     version: String,
     bspVersion: String,
     rootUri: Uri,
     capabilities: BuildClientCapabilities,
-    data: Option[Json]
+    data: Option[RawJson]
 )
 
-@JsonCodec final case class Reload()
+object InitializeBuildParams {
+  implicit val codec: JsonValueCodec[InitializeBuildParams] =
+    JsonCodecMaker.makeWithRequiredCollectionFields
+}
+
+object NullAllowingCodecWrapper {
+
+  def apply[A](codec: JsonValueCodec[A]): JsonValueCodec[A] =
+    new JsonValueCodec[A] {
+      override def decodeValue(in: JsonReader, default: A): A =
+        if (in.isNextToken('n')) {
+          in.readNullOrError[String]("x", "expected value or null")
+          nullValue
+        } else {
+          in.rollbackToken()
+          codec.decodeValue(in, default)
+        }
+
+      override def encodeValue(x: A, out: JsonWriter): Unit =
+        if (x == null) out.writeNull()
+        else codec.encodeValue(x, out)
+
+      override val nullValue: A = null.asInstanceOf[A]
+    }
+
+}
```

Comment:
This is one of the other things I needed to add to make the requests work that allowed for `null`, for example `Reload`. This is also a bit more generic just to allow for it also be used in the case where you actually want to optionally allow for `null`. However in our case the main thing we are using it for is to just ensure capture the null value. Thanks to Andriy on Gitter for helping me figure this out.